crates/hir/src/semantics/source_to_def.rs

```diff
@@ -106,7 +106,7 @@ use syntax::{
 
 use crate::{db::HirDatabase, InFile};
 
-pub(super) type SourceToDefCache = FxHashMap<ChildContainer, DynMap>;
+pub(super) type SourceToDefCache = FxHashMap<ChildContainer, FxHashMap<HirFileId, DynMap>>;
```

Comment:
I'd go for the compound keys if it's not too complicated.

Reply:
Oh right, I forgot to switch to that. Was more interested in seeing whether this passes the tests first.